src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WorkspaceSemanticTokensRefreshNotifier.cs

```diff
@@ -20,12 +21,10 @@ internal class WorkspaceSemanticTokensRefreshNotifier : IWorkspaceSemanticTokens
     private readonly CancellationTokenSource _disposeTokenSource;
     private readonly IDisposable _optionsChangeListener;
 
-    private readonly object _gate = new();
-    private bool? _supportsRefresh;
-    private bool _waitingToRefresh;
-    private Task _refreshTask = Task.CompletedTask;
+    private readonly AsyncBatchingWorkQueue _queue;
```

Comment:
Amusingly, I had used an AsyncBatchingWorkQueue in #10140 and removed it based on feedback.

Reply:
well I'm bringing it back! :P